### PR TITLE
Add standard command-line flags

### PR DESCRIPTION
**Pull request by chriskuehl**
#### no arguments
```
ckuehl@dev15-devc:~/proj/dumb-init$ ./dumb-init
Usage: ./dumb-init [option] program [args]
Try ./dumb-init --help for full usage.
```
#### --help / -h
```
ckuehl@dev15-devc:~/proj/dumb-init$ ./dumb-init --help
dumb-init v0.4.0
Usage: ./dumb-init [option] command [[arg] ...]

dumb-init is a simple process supervisor that forwards signals to children.
It is designed to run as PID1 in minimal container environments.

Optional arguments:
   -c, --single-child   Run in single-child mode.
                        In this mode, signals are only proxies to the
                        direct child and not any of its ancestors.
   -v, --verbose        Print debugging information to stderr.
   -h, --help           Print this help message and exit.
   -V, --version        Print the current version and exit.

Full help is available online at https://github.com/Yelp/dumb-init
ckuehl@dev15-devc:~/proj/dumb-init$ ./dumb-init -v
Usage: ./dumb-init [option] program [args]
Try ./dumb-init --help for full usage.
```
#### --version / -V
```
ckuehl@dev15-devc:~/proj/dumb-init$ ./dumb-init --version
dumb-init v0.4.0
```
#### --verbose / -v
```
ckuehl@dev15-devc:~/proj/dumb-init$ ./dumb-init -v echo hi
[dumb-init] Child spawned with PID 1392587.
[dumb-init] setsid complete.
hi
[dumb-init] A child with PID 1392587 exited with exit status 0.
[dumb-init] Forwarded signal 15 to children.
[dumb-init] Child exited with status 0. Goodbye.
```
#### --single-child --verbose / -cv
```
ckuehl@dev15-devc:~/proj/dumb-init$ ./dumb-init -cv echo hi
[dumb-init] Child spawned with PID 1392920.
hi
[dumb-init] A child with PID 1392920 exited with exit status 0.
[dumb-init] Forwarded signal 15 to children.
[dumb-init] Child exited with status 0. Goodbye.
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -48,9 +48,14 @@ DOCKER_TOX_TEST := sh -uexc ' \
...
 '
 .PHONY: build
-build:
+build: VERSION.h
 	$(CC) $(CFLAGS) -o dumb-init dumb-init.c
 
+VERSION.h: VERSION
+	echo '// THIS FILE IS AUTOMATICALLY GENERATED' > VERSION.h
+	echo '// Run `make VERSION.h` to update it after modifying VERSION.' >> VERSION.h
+	xxd -i VERSION >> VERSION.h
```
Comment (chriskuehl) [inline Makefile]: can't really decide if this is simpler than just duplicating the version in the C file or not...

unfortunately there doesn't seem to be a good way to have the C preprocessor read from a file into a string

Comment (bukzor) [inline Makefile]: I always side with no duplication of data.

File: VERSION.h [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// THIS FILE IS AUTOMATICALLY GENERATED
+// Run `make VERSION.h` to update it after modifying VERSION.
+unsigned char VERSION[] = {
+  0x30, 0x2e, 0x34, 0x2e, 0x30, 0x0a
```
Comment (bukzor) [inline VERSION.h]: lols

Comment (chriskuehl) [inline VERSION.h]: :(
I hate committing build artifacts but I don't know of a better way to handle this. The alternative is duplicate the version into the top of the C file. But then we need to maintain versions in sync in README/setup.py/dumb-init.c/debian changelog.

**Comment by bukzor** [comment]
The five cases you put in your description I'd actually do under test.

It sounds like a pain / duplication of data to do it like that, but its always been worth the cost to have the test suite tell me when I've derped up the messaging. Plus, even when it's doing what you meant to, often you only notice the output could be improved when you see it fully written out, under test.

But that's just me. Looks good apart from that nit. :shipit:

File: VERSION.h (hunk 2 above)
Comment (bukzor) [inline VERSION.h]: I just thought the ascii byte representation was funny is all.

**Pull request merged**